Cleanup of cluster-config.yaml.tpl (#4336)
kyma-project/kyma: docs/kyma/05-01-overview.md

type: Configuration
---

## Introduction

Users can configure Kyma installation in two ways:
  - Customize the list of the components to install
  - Provide overrides that change configuration values used by one or more components

The list of components to install is defined in the **Installation** Custom Resource.
Overrides are defined as ConfigMap or Secret objects defined by the user before the installation starts.
The Installer is the Kyma component that is responsible for reading and applying the configuration.

> ## Default settings

Reviewer: Consider splitting cluster and local sections with a toggle switch.

Author: Done.

type: Configuration
---

> 

Reviewer 2: @Tomasz-Smelcerz-SAP you've mentioned deletion of cluster config yaml but I cannot see it being deleted here?

Author: Removed.

Other overrides are component-specific.
To learn about configuration options for a given component, refer to **Configuration** section of the component's documentation.
Once you know the name and possible set of values for a configuration option, define an override for it by creating a ConfigMap or a Secret. You can also extend an existing one.
[Read more](#configuration-helm-overrides-for-kyma-installation) about the types of overrides and the rules for creating them.
>**CAUTION:** An override must exist in a cluster before the installation is started, otherwise the Installer is not be able to apply it correctly.

## Runtime configuration

Changing static configuration of the Kyma after installation is generally not supported.
Some components may support custom runtime configuration changes though.
For an example of such a runtime configuration, see [Helm Broker configuration](/components/helm-broker/#configuration-configuration) in which you add links to the bundle repositories to a ConfigMap and label it with the `helm-broker-repo=true` label for the Helm Broker to expose additional Service Classes in the Service Catalog.

> Another solution for changing components configuration after installation is to alter it's overrides and update the component using Kyma's [Update procedure](#installation-update-kyma).

Reviewer: ```suggestion
Another solution for changing components configuration after installation is to alter its overrides and update the component using Kyma's [Update procedure](#installation-update-kyma).
```

type: Configuration
---

> ## Introduction

Author: Done.

Users can configure Kyma installation in two ways:
  - Customize the list of the components to install
  - Provide overrides that change configuration values used by one or more components

The list of components to install is defined in the **Installation** Custom Resource.
Overrides are defined as ConfigMap or Secret objects defined by the user before the installation starts.
The Installer is the Kyma component that is responsible for reading and applying the configuration.

## Default settings

Default configuration is defined differently for cluster and local installations.

<div tabs>
  <details>
  <summary>
  Local installation
  </summary>

  The list of available components is stored in Kyma project sources in the file **kyma/installation/resources/installer-cr.yaml.tpl**
  Default installation overrides can be found in the file **kyma/installation/resources/installer-config-local.yaml.tpl**

>   All the other configuration values are defined directly in components in **/resources** subdirectory of Kyma project.

Reviewer 3: ```suggestion
  Other configuration values are defined directly in components in the configuration of the respective components.
```

  </details>
  <details>
  <summary>
  Cluster installation
  </summary>

  The installation is based on a released Kyma version.
  You can find the list of available components in the release artifact **kyma-installer-cluster.yaml**.
  Any necessary overrides are described in the [installation procedure](#installation-install-kyma-on-a-cluster).
  All the other configuration values are defined directly in components released with the Kyma version.
  </details>
</div>

## Installation configuration

Before you start the Kyma installation process, you can customize the default settings.

### Components

One of the released Kyma artifacts is the Kyma-Installer, a Docker image that combines the Installer executable with charts for all components available in the release.
The Kyma-Installer can install only the components contained in its image.
**Installation** Custom Resource specifies which components of all available ones should be actually installed.
Components that are not an integral part of the default Kyma Lite package are commented out with a hash character (#) and skipped during installation.

> You can customize list of components used for installation by:

Reviewer 3: ```suggestion
You can customize the list of components by:
```

> - Uncommenting component entry to enable a component installation.

Reviewer 3: ```suggestion
- Uncommenting a component entry to install the component.
```

> - Commenting out an entry using hash chacter to disable a component installation.

Reviewer 3: ```suggestion
- Commenting out a component entry using the hash character (#) to skip the installation of that component.
```

> - Adding new components to the list along with their chart definition. In that case you must create your own [Kyma-Installer image](#installation-use-your-own-kyma-installer-image) as you are adding new component to Kyma.

Reviewer 3: ```suggestion
- Adding new components along with their chart definitions to the list. If you do that, you must build your own [Kyma Installer image](#installation-use-your-own-kyma-installer-image) as you are adding a new component to Kyma.
```

For more details on custom component installation, see [this](#configuration-custom-component-installation) document.

### Overrides

> Common overrides that affect entire installation, for example `global.domainName`, are already described in the [installation procedure](#installation-overview).

Reviewer 3: ```suggestion
The common overrides that affect the entire installation, are described in the installation guides.
```

Other overrides are component-specific.

> To learn about configuration options for a given component, refer to **Configuration** section of the component's documentation.

Reviewer 3: ```suggestion
To learn more about the configuration options available for a specific component, see the **Configuration** section of the component's documentation.
```

> Once you know the name and possible set of values for a configuration option, define an override for it by creating a ConfigMap or a Secret. You can also extend an existing one.

Reviewer 3: This sentence is redundant IMO, I'd remove it.

Author: Done.

[Read more](#configuration-helm-overrides-for-kyma-installation) about the types of overrides and the rules for creating them.

> >**CAUTION:** An override must exist in a cluster before the installation is started, otherwise the Installer is not be able to apply it correctly.

Reviewer 3: ```suggestion
>**CAUTION:** An override must exist in a cluster before the installation starts. If you fail to deliver create the override before the installation, the configuration can't be applied. 
```

Reviewer 3: Please add a newline between the "Read more" sentence and the CAUTION panel.

> **CAUTION:** In documentation for each Kyma component, you can find configuration documents that list configurable parameters from the `values.yaml` files for a given component chart or sub-chart. Override values only for the parameters exposed in those configuration documents.

> ## Runtime configuration

Reviewer 3: Please remove this section

Author: Done.
